lib/l1_watcher/src/tx_watcher.rs

```diff
@@ -51,54 +48,19 @@ impl L1TxWatcher {
 
         tracing::info!(next_l1_block, "resolved on L1");
 
-        let l1_watcher = L1Watcher::new(zk_chain, next_l1_block, config.max_blocks_to_process);
-
-        Ok(Self {
-            l1_watcher,
+        let this = Self {
             next_l1_priority_id,
             output,
-            poll_interval: config.poll_interval,
-        })
-    }
+        };
+        let l1_watcher = L1Watcher::new(
```

Comment:
I guess we could also extract the init logic (including binsearch) from individual watchers to the base l1_watcher. Currently the control flow starts inside the subclasses - and then they pass it over to the base class, passing `this` as a parameter.
Can of course be done seaprately/later